agent/lib/kontena/workers/container_info_worker.rb

```diff
@@ -38,22 +44,26 @@ def on_container_event(topic, event)
 
       container = Docker::Container.get(event.id)
       if container
-        self.publish_info(container)
-        self.notify_coroner(container) if container.suspiciously_dead?
+        publish_info(container)
       end
     rescue Docker::Error::NotFoundError
-      self.publish_destroy_event(event)
+      publish_destroy_event(event)
     rescue => exc
       error "#{exc.class.name}: #{exc.message}"
       error exc.backtrace.join("\n")
     end
 
     def on_container_publish_info(topic, container)
-      self.publish_info(container)
+      publish_info(container)
     end
 
     def on_websocket_connected(topic, data)
-      self.publish_all_containers
+      publish_all_containers
+      container_coroner.start
+    end
+
+    def on_websocket_disconnect(topic, data)
+      container_coroner.stop
```

Comment:
Why do we need to stop the coroner... wouldn't it be better to just let it sit in its `report` retry-loop until the websocket reconnects? That way it can report on those destroyed containers that we missed out publishing info for?